docs/localnet/localnet.md

```diff
@@ -0,0 +1,380 @@
+
+```shell
+jq '.account_id = "sam.test.near"' ~/.near/mpc-sam/validator_key.json > ~/.near/mpc-sam/temp.json && mv ~/.near/mpc-sam/temp.json ~/.near/mpc-sam/validator_key.json
+```
+
+Create a `config.yaml` for the MPC-indexer:
+
+```bash
+cat > ~/.near/mpc-sam/config.yaml << 'EOF'
+my_near_account_id: sam.test.near
+near_responder_account_id: sam.test.near
+number_of_responder_keys: 1
+web_ui:
+  host: localhost
+  port: 8082
+triple:
+  concurrency: 2
+  desired_triples_to_buffer: 128
+  timeout_sec: 60
+  parallel_triple_generation_stagger_time_sec: 1
+presignature:
+  concurrency: 4
+  desired_presignatures_to_buffer: 64
+  timeout_sec: 60
+signature:
+  timeout_sec: 60
+indexer:
+  validate_genesis: false
+  sync_mode: Latest
+  concurrency: 1
+  mpc_contract_id: mpc-contract.test.near
+  finality: optimistic
+ckd:
+  timeout_sec: 60
+cores: 4
+EOF
+```
+
+### Run the MPC binary
+
+In two separate shells run the MPC binary for frodo and sam. Note the last argument repeating (`11111111111111111111111111111111`) is the encryption key for the secret storage, and can be any arbitrary value.
+
+```shell
+mpc-node start --home-dir ~/.near/mpc-sam/ 11111111111111111111111111111111 --image-hash "8b40f81f77b8c22d6c777a6e14d307a1d11cb55ab83541fbb8575d02d86a74b0" --latest-allowed-hash-file /temp/LATEST_ALLOWED_HASH_FILE.txt local
+```
+
+```shell
+mpc-node start --home-dir ~/.near/mpc-frodo/ 11111111111111111111111111111111 --image-hash "8b40f81f77b8c22d6c777a6e14d307a1d11cb55ab83541fbb8575d02d86a74b0" --latest-allowed-hash-file /temp/LATEST_ALLOWED_HASH_FILE.txt local
+```
+Note: `8b40f81f77b8c22d6c777a6e14d307a1d11cb55ab83541fbb8575d02d86a74b0` is just an arbitrary hash.
+
+In the shell where you ran the local near node, you should see the peer count change from 0 to 2 as the frodo and sam MPC indexers connect to it.
+
+```log
+2025-08-03T14:19:42.179075Z  INFO stats: #  100530 Fe9M4GuFpnTgMJvwZR1uzsxMAp7gKqWp1GAVdm5RY5Rc Validator | 1 validator 0 peers ⬇ 0 B/s ⬆ 0 B/s 1.70 bps 0 gas/s CPU: 3%, Mem: 2.17 GB
+2025-08-03T14:19:52.181980Z  INFO stats: #  100546 G9tp5Jwh5pfreNqREKJT75dgq6Zx6w4hXtyecN4r4rWC Validator | 1 validator 0 peers ⬇ 0 B/s ⬆ 0 B/s 1.60 bps 0 gas/s CPU: 3%, Mem: 2.17 GB
+2025-08-03T14:20:02.182157Z  INFO stats: #  100563 DqWZMLg9e7Z55u3JxjUBBvjUCJf1T3K4K4XWWbhQfG6a Validator | 1 validator 2 peers ⬇ 69 B/s ⬆ 238 B/s 1.70 bps 0 gas/s CPU: 1%, Mem: 2.17 GB
+2025-08-03T14:20:12.183398Z  INFO stats: #  100579 98DQh3yG987rY1pNWKbM4jYjJ5xuFixP4g3MJuVvpiWY Validator | 1 validator 2 peers ⬇ 1.10 kB/s ⬆ 37.4 kB/s 1.60 bps 0 gas/s CPU: 3%, Mem: 2.17 GB
+```
+
+### 5. Assign the signer and responder keys as subkeys.
+
+We must delegate the generate signing keys Sam and Frodo generated as access keys to their near accounts such that they
+can sign transaction that require authorization on the contract.
+
+```shell
+docs/localnet/scripts/assign_access_keys.sh frodo 8081
+```
```

Comment:
Note: While this gets the job done, I want this guide to be pretty low level and expose as much of the raw commands to the reader. We could write scripts to automate this whole setup, which we kind of already have in our pytests.

So I'll update this and some of the other scripts to follow the patterns of this guide with raw commands.